src/main/java/com/example/pokebattlez/battle/model/Pokemon.java

```diff
@@ -0,0 +1,83 @@
+package com.example.pokebattlez.battle.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import static com.example.pokebattlez.battle.model.Stat.*;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Pokemon {
+    private static final float NATURE_MODIFIER = 0.1f;
+
+    private String name;
+    private int position;
+
+    private int level;
+
+    private Map<com.example.pokebattlez.battle.model.Stat, Stat> stats = Map.of(
+            HP, new Stat(Stat::calculateHp),
+            ATTACK, new Stat(),
+            DEFENCE, new Stat(),
+            SP_ATTACK, new Stat(),
+            SP_DEFENCE, new Stat(),
+            SPEED, new Stat()
+    );
+
+    private String gender;
+    private String nature;
+    private String heldItem;
+    private String ability;
+
+    private String move1;
+    private String move2;
+    private String move3;
+    private String move4;
+
+    public void setNature(Nature nature) {
+        stats.get(nature.getUp()).modifyNature(NATURE_MODIFIER);
+        stats.get(nature.getDown()).modifyNature(-NATURE_MODIFIER);
+    }
+
+    @Data
+    @NoArgsConstructor
+    public static class Stat {
```

Comment:
Why is this class withing the Pokemon. Isn't it a general Stat type?

Reply:
Stat instances are only ever gonna exist inside of a Pokemon instance. They should never be standalone.